sources/TileDB.CSharp/Marshalling/ScratchBuffer.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Buffers;
+using System.Runtime.InteropServices;
+
+#nullable enable
+
+namespace TileDB.CSharp.Marshalling
+{
+    /// <summary>
+    /// Manages temporary buffers with low GC pressure.
+    /// </summary>
+    /// <remarks>
+    /// Objects of this type should not be passed around; instead pass their <see cref="Span"/> property.
+    /// </remarks>
+    /// <typeparam name="T">The type of items this scratch buffer stores.</typeparam>
+    internal ref struct ScratchBuffer<T>
```

Comment:
Copied from https://gist.github.com/teo-tsirpanis/db9ae014efcad311a5df87e6ed04bc51. I can strip license headers, it's mine and in public domain.